src/core/undo/MovePageUndoAction.cpp

```diff
@@ -0,0 +1,28 @@
+#include "MovePageUndoAction.h"
+
+#include "util/i18n.h"              // for _
+
+MovePageUndoAction::MovePageUndoAction(const PageRef& page, size_t oldPos, size_t newPos):
+        UndoAction("MovePageUndoAction"),
+        oldPos(page, newPos, true),
+        newPos(page, oldPos, false) {
+    this->page = page;
+}
+
+MovePageUndoAction::~MovePageUndoAction() { this->page = nullptr; }
+
+auto MovePageUndoAction::undo(Control* control) -> bool {
+    bool oldRes = oldPos.undo(control);
+    bool newRes = newPos.undo(control);
```

Comment:
I'm a bit confused here. Is this reinserting the page at its old position first and then removing it from the new one? Should itt be the other way around?

Also, more explicit names for oldPos and newPos could help.